lms/djangoapps/courseware/views/views.py

```diff
@@ -860,18 +860,17 @@ def _progress(request, course_key, student_id):
         except User.DoesNotExist:
             raise Http404
 
-    # NOTE: To make sure impersonation by instructor works, use
-    # student instead of request.user in the rest of the function.
-
     # The pre-fetching of groups is done to make auth checks not require an
     # additional DB lookup (this kills the Progress page in particular).
     student = User.objects.prefetch_related("groups").get(id=student.id)
-
-    if student_id != student.id:
+    if request.user.id != student.id:
```

Comment:
This change is why the dark launch python tests were failing. There's a difference between masquerading (`student_id` is None, request.user is the masquerade target) and a staff user getting a student's progress (student_id is the target, request.user is the staff originator). That test was checking the latter situation, which was failing to re-`get_course_with_access` as `student.id` *always* == `student__id` in that scenario.